src/services/blockchain/ethereum/ethereum-blockchain-service.ts

```diff
@@ -89,14 +95,24 @@ export default class EthereumBlockchainService implements BlockchainService {
       txData.gasLimit = BigNumber.from(this.config.blockchain.connectors.ethereum.gasLimit)
       logger.debug('Overriding Gas limit: ' + txData.gasLimit.toString())
     } else {
-      const gasPriceEstimate = await this.wallet.provider.getGasPrice() // in wei
+      const feeData = await this.wallet.provider.getFeeData()
       // Add extra to gas price for each subsequent attempt
-      txData.gasPrice = EthereumBlockchainService.increaseGasPricePerAttempt(
-        gasPriceEstimate,
+      const prevMaxPriorityFeePerGas = BigNumber.from(txData.maxPriorityFeePerGas || 0)
+      const nextMaxPriorityFeePerGas = EthereumBlockchainService.increaseGasPricePerAttempt(
+        feeData,
         attempt,
-        txData.gasPrice
+        prevMaxPriorityFeePerGas
+      )
+      const difference = nextMaxPriorityFeePerGas.sub(prevMaxPriorityFeePerGas)
+      if (feeData.maxPriorityFeePerGas) {
```

Comment:
saw your comment below.  Is this also about working Ganache not supporting EIP1559?  I hate having production code that exists just to work around testing, if it can be avoided (although admittedly sometimes it can't be avoided).  Would it be possible to introduce something in the test setup that wraps Ganache and coerces it to an EIP1559-compatible API, so that the workaround lives entirely in test code not in production code?

Reply:
There is a [PR in the works](https://github.com/trufflesuite/ganache/pull/1135) for Ganache to support 1559, so this thing is not for long.

And, no, there is no way I can imagine to wrap Ganache into something with 1559 support.

Reply:
feels like it wouldn't be that hard to mock the `getFeeData()` and `sendTransaction()` functions to coerce the txData into the right format?  i.e. making `getFeeData()` replace the `gasPrice` field with the `maxPriorityFeePerGas` field, and `sendTransaction()` replace the `maxPriorityFeePerGas` field with a `gasPrice` field.  Basically just doing the same thing as we are doing here and in `increaseGasPricePerAttempt` to accommodate non-1559 transactions, but moved into test code that injects a mock that intercepts the messages before passing them on?  That would put all the non-1559 support code squarely within the test file and allow the production code to only worry about the formats for 1559 transactions.

If there's something that makes this harder than I'm realizing, then at the very least we should add TODO comments to mark every line in the production code that only exists to support Ganache so it'll be clear when we can rip them out.